### PR TITLE
Movie trailer branch

### LINKED ISSUE #84: Add Movie / TV Show videos in the detail fragments
### Suggestion / Feature Request

Display the a list of different trailers and videos in the detail fragments as a horizontal pager where one can swipe or press on an arrow button (your choice which implementation you wanna go with, or you can implement both) to move to the next video

You can use [`YoutubeView`](https://github.com/PEC-CSS/MovieDroid/blob/main/app/src/main/java/com/pec_acm/moviedroid/mainpage/YoutubeView.kt)  in the layout to display a video, usage in kt file to display a video->
```kt
youtubeView.loadData(key, "text/html", "utf-8")
```
The videos and their keys can be obtained using [this api endpoint](https://developers.themoviedb.org/3/movies/get-movie-videos)

For now we'll stick ourselves to only youtube videos so ignore the videos which have `site` other than `"Youtube"`

### PR DESCRIPTION
**Pull request by krishnachaitanya0107**
## Description

- Implemented functionality to get movie/tv trailers from [API endpoint](https://developers.themoviedb.org/3/movies/get-movie-videos) 
- Created a horizontal recyclerview with youtubeview items to display trailers/videos

## Issue Resolved

Fixes #84 

## Video

https://user-images.githubusercontent.com/76201176/194746142-03bbc904-3658-40f5-86be-c9e9929d471d.mp4

### CODE REVIEW

File: app/src/main/java/com/pec_acm/moviedroid/data/api/TMDBApi.kt [hunk 1]
```diff
@@ -90,4 +87,12 @@ interface TMDBApi {
...
         apiKey : String = Constants.API_KEY,
     ) : Response<TVDetail>
+
+    @GET("movie/{movie_id}/videos")
+    suspend fun movieTvVideosByID(
+        @Path("movie_id") movie_id : Int,
+        @Query("api_key")
+        apiKey : String = Constants.API_KEY
+    ) : Response<MovieTvVideo>
+
```
Comment (ken1000minus7) [inline app/src/main/java/com/pec_acm/moviedroid/data/api/TMDBApi.kt]: This endpoint is only for movies, there is a separate endpoint for [tv shows](https://developers.themoviedb.org/3/tv/get-tv-videos). Just create a separate function for that and make changes in other files accordingly, also change the names of the existing functions accordingly indicating that they are for movies and not tv shows 😊

**Review by ken1000minus7 — CHANGES_REQUESTED** [review]
Great work @krishnachaitanya0107 😄 , everything is almost perfect but there are a few changes required 😇

File: app/src/main/java/com/pec_acm/moviedroid/data/api/TMDBApi.kt (hunk 1 above)
Comment (krishnachaitanya0107) [reply app/src/main/java/com/pec_acm/moviedroid/data/api/TMDBApi.kt]: Yeah good catch , 

actually i noticed that there was a different endpoint for tv shows and the thing is i reused the same apiRespose data class , but forgot to create a different get method for the tv endpoint 😅 . Will fix that in a bit .

Thanks for confirming 👍

**Comment by krishnachaitanya0107** [comment]
Hi @ken1000minus7 , 

made the required changes. Please go through it.

Thanks

**Review by ken1000minus7 — APPROVED** [review]
Great work @krishnachaitanya0107 🚀 , lgtm, thanks for your contribution 😄

**Comment by krishnachaitanya0107** [comment]
Thank you for allowing me to work on this issue

**Pull request merged**